pkg/envoy/server.go

```diff
@@ -498,14 +499,16 @@ func (s *XDSServer) UpdateNetworkPolicy(ep logger.EndpointUpdater, policy *polic
 	ingressPolicyEnforced, egressPolicyEnforced bool, labelsMap identity.IdentityCache,
 	deniedIngressIdentities, deniedEgressIdentities map[identity.NumericIdentity]bool, wg *completion.WaitGroup) error {
 
-	s.mutex.Lock()
+	s.mutex.RLock()
+
 	// If there are no listeners configured, the local node's Envoy proxy won't
 	// query for network policies and therefore will never ACK them, and we'd
 	// wait forever.
 	if !viper.GetBool("sidecar-http-proxy") && len(s.listeners) == 0 {
 		wg = nil
 	}
-	s.mutex.Unlock()
+
+	s.mutex.RUnlock()
```

Comment:
Do we need two separate critical sections here? Is it possible for something else to remove a listener in between?

Reply:
I don't think it would matter much.